addons/mail/static/src/utils/test_utils.js

```diff
@@ -257,7 +254,99 @@ async function beforeEach(self) {
         }
         data[modelName] = { fields, records };
     }
+    return data;
+}
+
+let pyEnv;
+/**
+ * Creates an environment that can be used to setup test data as well as
+ * creating data after test start.
+ *
+ * @returns {Object} An environment that can be used to interact with the mock
+ * server (creation, deletion, update of records...)
+ */
+ export async function startServer() {
+    pyEnv = new Proxy(
+        {
+            mockServer: new MockServer(await getInitialServerData(), {}),
+            ...TEST_USER_IDS,
+        },
+        {
+            get(target, name) {
+                if (target[name]) {
+                    return target[name];
+                }
+                return {
+                    /**
+                     * Simulate a 'create' operation on a model.
+                     *
+                     * @param {Object[] | Object} values records to be created.
+                     * @returns {Number[] | Number} array of ids if more than one value was passed,
```

Comment:
just nit-picking:
- no whitespace around `|`
- we prefer being more specific than `number` by using either `integer` or `float`
- on uppercase for `integer`/`boolean`/`float` because these are primitive types. Keep uppercase for custom classes/types and also for `Array` and `Object`.